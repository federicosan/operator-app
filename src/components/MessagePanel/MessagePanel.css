#message-panel {
  flex-grow: 3;
}