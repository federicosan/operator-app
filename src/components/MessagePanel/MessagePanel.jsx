import React, { PropTypes } from 'react';
import { connect } from 'react-redux';

import MessageMenuBar from '../MessageMenuBar/MessageMenuBar.jsx';
import MessageList from '../MessageList/MessageList.jsx';
import InputBar from '../InputBar/InputBar.jsx';
import './MessagePanel.css';

const MessagePanel = (props) => {
  const { activeId } = props;
  const renderView = () => {
    if (!activeId) {
      return (
        <div className="MessagePanel panel">
          <MessageList />
        </div>
      );
    }

    return (
      <div className="MessagePanel panel">
        <MessageMenuBar />
        <MessageList />
        <InputBar />
      </div>
    );
  };

  return renderView();
};

MessageList.propTypes = {
  activeId: PropTypes.string,
};


const mapStateToProps = state => ({
  activeId: state.chat.activeId,
});

const mapDispatchToProps = dispatch => ({
  dispatch,
});


export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(MessagePanel);
