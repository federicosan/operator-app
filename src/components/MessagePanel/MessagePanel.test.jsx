import React from 'react';

import { shallow } from 'enzyme';

import MessagePanel from './MessagePanel.jsx';

const store = {
  subscribe: jest.fn(),
  dispatch: jest.fn(),
  getState: jest.fn(() => ({ })),
};

describe('MessagePanel', () => {
  it('matches snapshot', () => {
    const component = shallow(<MessagePanel store={store} />);

    expect(component).toMatchSnapshot();
  });
});
