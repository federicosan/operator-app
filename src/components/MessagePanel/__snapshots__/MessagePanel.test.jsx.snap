exports[`MessagePanel matches snapshot 1`] = `
<MessagePanel
  dispatch={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "subscribe": [Function],
    }
  } />
`;
